app/controllers/admin/loan_response_sets_controller.rb

```diff
@@ -1,22 +1,39 @@
 class Admin::LoanResponseSetsController < Admin::AdminController
+  include QuestionnaireRenderable
+
   def create
-    @record = LoanResponseSet.new(record_params)
-    authorize @record
-    @record.save!
+    @response_set = LoanResponseSet.new(response_set_params)
+    authorize @response_set
+    @response_set.save!
     redirect_to display_path, notice: I18n.t(:notice_created)
   end
 
   def update
-    @record = LoanResponseSet.find(params[:id])
-    authorize @record
-    @record.update!(record_params)
-    redirect_to display_path, notice: I18n.t(:notice_updated)
+    @response_set = LoanResponseSet.find(params[:id])
+    authorize @response_set
+    [:updater, :updated_at, :lock_version].each { |i| instance_variable_set "@#{i}", @response_set.send(i) }
```

Comment:
The reason is @response_set gets modified later, and we need the version from the database. I'll clarify.